### PR TITLE
[darwin] Update pixel format handling in FlutterTexture

### PR DESCRIPTION
**Pull request by bc-lee**
CVPixelBuffers from the application can have arbitrary pixel formats. However, current implementation doesn't support all pixel formats. To address this, add a comment to the FlutterTexture class to clarify which pixel formats are supported. Also, reject unsupported pixel formats so that the application can handle them properly.

Fixes [#147242](https://github.com/flutter/flutter/issues/147242).

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/darwin/graphics/FlutterDarwinExternalTextureMetal.mm [hunk 1]
```diff
@@ -133,8 +133,11 @@ - (void)onTextureUnregistered {
   if (_pixelFormat == kCVPixelFormatType_420YpCbCr8BiPlanarVideoRange ||
       _pixelFormat == kCVPixelFormatType_420YpCbCr8BiPlanarFullRange) {
     image = [self wrapNV12ExternalPixelBuffer:pixelBuffer context:context];
-  } else {
+  } else if (_pixelFormat == kCVPixelFormatType_32BGRA) {
     image = [self wrapRGBAExternalPixelBuffer:pixelBuffer context:context];
```
Comment (chinmaygarde) [inline shell/platform/darwin/graphics/FlutterDarwinExternalTextureMetal.mm:137]: This method is incorrectly named. Perhaps rename to `wrapBGRAExternalPixelBuffer:context:` too? Internally, it does assume BGRA as well.

Comment (bc-lee) [reply shell/platform/darwin/graphics/FlutterDarwinExternalTextureMetal.mm]: Done. Note that `[FlutterDarwinExternalTextureSkImageWrapper: wrapRGBATexture:grContext:width:height:]` also has RGBA in its name. I'm not sure renaming all of them to use BGRA is a good idea, but I'm open to suggestions.

Comment (chinmaygarde) [reply shell/platform/darwin/graphics/FlutterDarwinExternalTextureMetal.mm]: If its not already a public method and the name is more fitting, go for it!

Comment (bc-lee) [reply shell/platform/darwin/graphics/FlutterDarwinExternalTextureMetal.mm]: I was considering renaming the `wrapRGBATexture` method, but I think it is better to keep the method name as it is.

[Link1](https://github.com/flutter/engine/blob/8578edf9c9393471ca9eab18e9154f0e6066dcb6/shell/platform/embedder/embedder.h#L602)                                                                                                                      [Link2](https://github.com/flutter/engine/blob/8578edf9c9393471ca9eab18e9154f0e6066dcb6/shell/platform/embedder/embedder_external_texture_metal.mm#L73)

We have `kYUVA` and `kRGBA` as the two possible values for `FlutterMetalExternalTexturePixelFormat`, and the function I mentioned above is used along with the enum, `FlutterMetalExternalTexturePixelFormat`. So it sounds like renaming would be more confusing.

**Review by chinmaygarde — APPROVED** [review]

**Comment by bc-lee** [comment]
Fell free to merge this PR. Thanks for reviewing!

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Pull request merged**